scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ByteCompressionUtils.java

```diff
@@ -0,0 +1,61 @@
+package org.ow2.proactive.scheduler.util;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+import org.apache.log4j.Logger;
+
+/**
+ * 
+ * @author ActiveEon team
+ *
+ */
+public class ByteCompressionUtils {
+	private static final Logger LOG = Logger.getLogger(ByteCompressionUtils.class);  
+    
+    
+	  public static byte[] compress(byte[] data) throws IOException {  
+	   Deflater deflater = new Deflater();  
+	   deflater.setInput(data);  
+	   
+	   ByteArrayOutputStream outputStream = new ByteArrayOutputStream(data.length);   
+	       
+	   deflater.finish();  
+	   byte[] buffer = new byte[1024];   
+	   while (!deflater.finished()) {  
+	    int count = deflater.deflate(buffer); // returns the generated code... index  
+	    outputStream.write(buffer, 0, count);   
+	   }  
+	   outputStream.close();  
+	   byte[] output = outputStream.toByteArray();  
+	   
+	   deflater.end();
+
+	   LOG.debug("Original: " + data.length / 1024 + " Kb");  
+	   LOG.debug("Compressed: " + output.length / 1024 + " Kb");  
+	   return output;  
+	  }  
+	   
+	  public static byte[] decompress(byte[] data) throws IOException, DataFormatException {  
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.util.zip.DataFormatException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)